torch/csrc/distributed/c10d/ProcessGroup.cpp

```diff
@@ -7,13 +7,23 @@
 namespace c10d {
 namespace {
 
-c10::intrusive_ptr<ProcessGroup::Work> broadcast(const c10::intrusive_ptr<ProcessGroup>& process_group,
-    at::TensorList tensors, int64_t root_rank, int64_t root_tensor, int64_t timeout) {
+c10::intrusive_ptr<ProcessGroup::Work> broadcast(at::TensorList tensors,
+    const c10::intrusive_ptr<ProcessGroup>& process_group, int64_t root_rank, int64_t root_tensor, int64_t timeout) {
   auto tensor_vec = tensors.vec();
   return process_group->broadcast_impl(tensor_vec,
       BroadcastOptions {root_rank, root_tensor, std::chrono::milliseconds(timeout)});
 }
 
+// Added specifically for LazyTensor.
```

Comment:
We definitely need a better API design here. Ideally, a single tensor input and output version of broadcast is appreciated. It can be either inplace or functional.

Reply:
why is single-tensor important for LT?